### ISSUE TITLE
Feature request: custom sorting function for tags

### ISSUE THREAD

**Issue by seebi** [squidfunk/mkdocs-material#4549]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

When using tags with different capitalisation, the ordering is wrong.

### Expected behaviour

Given these 3 tags - I expect the following order:

- BestPractice
- cmemc
- Configuration

### Actual behaviour

Given these 3 tags - this is the actual order:

- BestPractice
- Configuration
- cmemc

### Steps to reproduce

1. use tags with different capitalisation
2. go to the tag overview page

### Package versions

- Python: `Python 3.10.8`
- MkDocs: `mkdocs, version 1.3.1 from ../documentation.eccenca.com/.venv/lib/python3.10/site-packages/mkdocs (Python 3.10)`
- Material: `Version: 8.5.4+insiders.4.24.1`

### Configuration

```yaml
https://github.com/eccenca/documentation.eccenca.com/blob/main/mkdocs.yml#L33

-> https://documentation.eccenca.com/22.1/tags/
```

### System information

- Operating system: macos + latest ubuntu (github action)
- Browser: Chrome Version 107.0.5304.62 (Official Build) (arm64)

**Comment by facelessuser** [comment]
@squidfunk

```python
>>> unsorted_list = ["BestPractice", "Configuration", "cmemc"]
>>> sorted_list = sorted(unsorted_list, key=str.casefold)
>>> unsorted_list
['BestPractice', 'Configuration', 'cmemc']
>>> sorted_list
['BestPractice', 'cmemc', 'Configuration']
```

**Comment by squidfunk** [comment]
@facelessuser thanks, but I meant from `mkdocs.yml` via `!!python...`. However, I already figured it out, so nevermind 😉

**Comment by facelessuser** [comment]
🤦🏻 glad you figured it out

**Comment by squidfunk** [comment]
Support for tag compare function added in squidfunk/mkdocs-material-insiders@479287d43582e67dff909cf21027439584686e94. Updated docs in 4c8e6532.

**Comment by seebi** [comment]
Thanks for this extension.

ATM I can not validate it on our site, since we need to upgrade to mkdocs 1.4.x first - which is not as easy as I thought.

So do not wait for us on closing this issue please.

**Comment by seebi** [comment]
ok, I managed to upgrade and test this. Unfortunately, I got this error:

```
ERROR    -  Error reading page 'tutorials/index.md': 'tuple' object has no attribute 'casefold'
Traceback (most recent call last):
  File "/Users/seebi/Repositories/documentation.eccenca.com/.venv/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/Users/seebi/Repositories/documentation.eccenca.com/.venv/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/seebi/Repositories/documentation.eccenca.com/.venv/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/seebi/Repositories/documentation.eccenca.com/.venv/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/seebi/Repositories/documentation.eccenca.com/.venv/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/seebi/Repositories/documentation.eccenca.com/.venv/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/seebi/Repositories/documentation.eccenca.com/.venv/lib/python3.10/site-packages/mkdocs/__main__.py", line 234, in serve_command
    serve.serve(dev_addr=dev_addr, livereload=livereload, watch=watch, **kwargs)
  File "/Users/seebi/Repositories/documentation.eccenca.com/.venv/lib/python3.10/site-packages/mkdocs/commands/serve.py", line 83, in serve
    builder(config)
  File "/Users/seebi/Repositories/documentation.eccenca.com/.venv/lib/python3.10/site-packages/mkdocs/commands/serve.py", line 76, in builder
    build(config, live_server=live_server, dirty=dirty)
  File "/Users/seebi/Repositories/documentation.eccenca.com/.venv/lib/python3.10/site-packages/mkdocs/commands/build.py", line 308, in build
    _populate_page(file.page, config, files, dirty)
  File "/Users/seebi/Repositories/documentation.eccenca.com/.venv/lib/python3.10/site-packages/mkdocs/commands/build.py", line 177, in _populate_page
    page.markdown = config.plugins.run_event(
  File "/Users/seebi/Repositories/documentation.eccenca.com/.venv/lib/python3.10/site-packages/mkdocs/plugins.py", line 520, in run_event
    result = method(item, **kwargs)
  File "/Users/seebi/Repositories/documentation.eccenca.com/.venv/lib/python3.10/site-packages/material/plugins/tags/plugin.py", line 115, in on_page_markdown
    return self._render_tag_index(markdown, page, included)
  File "/Users/seebi/Repositories/documentation.eccenca.com/.venv/lib/python3.10/site-packages/material/plugins/tags/plugin.py", line 170, in _render_tag_index
    for args in sorted(
  File "/Users/seebi/Repositories/documentation.eccenca.com/.venv/lib/python3.10/site-packages/material/plugins/tags/plugin.py", line 39, in casefold
    return tag.casefold()
AttributeError: 'tuple' object has no attribute 'casefold'
task: Failed to run task "serve": exit status 1
```

Our updated versions:

- Python: `Python 3.10.8`
- MkDocs: `mkdocs, version 1.4.2 from .../.venv/lib/python3.10/site-packages/mkdocs (Python 3.10)`
- Material: `Version: 8.5.7+insiders.4.26.1`

but I build included this git commit 479287d

This corresponds to this page which created via `tags_extra_files`

- config: https://github.com/eccenca/documentation.eccenca.com/blob/main/mkdocs.yml#L257

**Comment by seebi** [comment]
Update: removing the `tags_extra_files` also raise the error (but on the main tags page):

`ERROR    -  Error reading page 'tags.md': 'tuple' object has no attribute 'casefold'`

**Comment by squidfunk** [comment]
Please provide a self-contained reproduction and attach it here as a zip. Thank you.

**Comment by seebi** [comment]
sure: 
[mkdocs-material-issues-4549.zip](https://github.com/squidfunk/mkdocs-material/files/9930759/mkdocs-material-issues-4549.zip)

**Comment by squidfunk** [comment]
Thanks for the quick reproduction, that's indeed related to the tags extra files. I'll look into it.

**Comment by squidfunk** [comment]
Released as part of 8.5.8+insiders-4.26.2.
